pytomorrowio/pytomorrowio.py

```diff
@@ -182,34 +187,38 @@ def available_fields(
 
     @staticmethod
     def _get_url() -> str:
+        """Return base URL for API requests."""
         # This method is required for test mocks
         return BASE_URL_V4
 
     async def _call_api(self, params: Dict[str, Any]) -> Dict[str, Any]:
+        """Call tomorrow.io API."""
         if self._session:
-            return await self._make_call(params, self._session)
+            return await self.__call_api(params, self._session)
 
         async with ClientSession() as session:
-            return await self._make_call(params, session)
+            return await self.__call_api(params, session)
 
-    async def _make_call(
+    async def __call_api(
         self, params: Dict[str, Any], session: ClientSession
     ) -> Dict[str, Any]:
+        """Make API call with session."""
         if self._remaining_requests_in_second == 0:
             await asyncio.sleep(1)
 
         try:
             resp = await session.post(
                 self._get_url(),
                 headers=self._headers,
-                data=json.dumps({**self._params, **params}),
+                json={**self._params, **params},
+                compress=False,
             )
-            resp_json = await resp.json()
+            resp_json = await resp.json(content_type=None)
         except ClientConnectionError as error:
             raise CantConnectException() from error
 
         self._rate_limits = CIMultiDict(
-            {k: int(v) for k, v in resp.headers.items() if "ratelimit" in k.lower()}
+            {k: int(v) for k, v in resp.headers.items() if "limit" in k.lower()}
```

Comment:
Is this change necessary?

Reply:
no, this was another thing I was changing while testing the issues reported in https://github.com/home-assistant/core/issues/77355